File: Back-flask/db/canteen.py
from db.db import Config
from datetime import datetime
class canteemDb:


    def __init__(self):
        self.conn=Config.connection
        self.cursor=self.conn.cursor()
    


    def add_centry(self,token_no,shift_c,date):
        print(date)
        qr=f"INSERT INTO canteen (token_no,shift_c,shift_date) VALUES('{token_no}',{shift_c},'{date}')"
        print(qr)
        self.cursor.execute(qr)
        print("ddd")
        self.conn.commit()


    def get_canteen(self):
        qr="SELECT TOP 100 * FROM canteen ORDER BY shift_date DESC"
        self.cursor.execute(qr)
        dd=self.cursor.fetchall()
        res=[]
        for i in dd:
            res.append({"tk_no":i.token_no,"shift_code":i.shift_c,"shift_date":i.shift_date.strftime('%Y-%m-%d %H:%M:%S')})
        return res
    
    def del_centry(self,shift_date):
        qr=f"DELETE FROM canteen WHERE shift_date='{shift_date}'"
        self.cursor.execute(qr)
        if self.cursor.rowcount==0:
            return False
        else:
            self.conn.commit()
            return True
        
    def get_report(self,sd,ed):
        
        qr=f"select c.token_no,NAME,cc.CATE,count(c.token_no) as time from canteen c,canteen_user cc where c.token_no=cc.TOKEN_NO and c.shift_date between '{sd}' and '{ed} 23:50:12'group by c.token_no,cc.NAME,cc.CATE order by cc.CATE"
        self.cursor.execute(qr)
        dd=self.cursor.fetchall()
        res=[]
        for i in dd:
            res.append({"tk_no":i.token_no,"name":i.NAME,"cate":i.CATE,"Count":i.time})
        return res
    
# kk=canteemDb()
# kk.add_centry(51254,1)
# # kk.del_centry('')
# print(kk.get_canteen())
    

# kk=canteemDb()
# print(kk.get_report('2023-12-01','2023-12-30'))

File: Backend/bio/canteen.js
const express=require('express')
const rs=express.Router()
const sql =require('mssql')

const config=require('./db')
//add user for canteen 
const pool = new sql.ConnectionPool(config);
const poolConnect = pool.connect(); 
const dbf = require('dbf');

rs.get("/c_entry",(req,res)=>{
  const pc = pool.request(); 
  pc.query('select  top 200 * from canteen  order by f_time desc ')
  .then((result)=>{
    res.send(result.recordsets[0])
  })
  .catch((err)=>{
        res.send({"error":err})
  })
  // let dateObj = new Date();
 
// // Subtract one day from current time                        
// dateObj.setDate(dateObj.getDate() - 1);
 
// console.log(dateObj.toJSON().slice(0,10));
//   res.send(dateObj.toJSON().slice(0,10));
})


rs.post("/c_entry",(req,res)=>{
  const pc = pool.request(); 
  pc.input('tk',sql.NVarChar,req.body.TOKEN_NO)
  .input('shift_c',sql.Int,req.body.SHIFT_C)
  .input('shift_date',sql.DateTime,req.body.SHIFT_DATE)
  .query('INSERT INTO CANTEEN (TOKEN_NO,SHIFT_C,SHIFT_DATE) VALUES(@tk,@shift_c,@shift_date)')
  .then((result)=>{
    send({status:200,data:"added"});
  })
  .catch((err)=>{
    res.send({status:400,data:"faild to insert"});
  })
})

// rs.delete("/c_entry",(req,res)=>{
//   const pc=pool.request();
//   pc.input()
// })



   




module.exports =rs